configmaster/management/handlers/config_backup.py

```diff
@@ -128,7 +131,11 @@ def _compare_config_checksum(self):
         :attr:`configmaster.models.Device.last_checksum` field if necessary.
         """
 
-        checksum = self._get_config_checksum()
+        try:
+            checksum = self._get_config_checksum()
+        except Exception as e:
+            logger.exception(e)
```

Comment:
But that has negative consequences, too - unnecessary commits will be made and the operations team gets spammed with change alerts.

I'd rather have intermittent failures than intermittent change alerts.

Unless this is a common occurrence, I don't think this warrants any extra code. We're already retrying tasks five times by default, if it *still* fails, something's clearly wrong.

If you insist on keeping it, the logging configuration will need to be documented/added to the project default settings. The config you suggested isn't terribly useful, since it would override the default config (which sends a mail each time an unhandled exception occurs). Nobody looks at the console output logs.

Reply:
> But that has negative consequences, too - unnecessary commits will be made and the operations team gets spammed with change alerts.
> 
> I'd rather have intermittent failures than intermittent change alerts.

I think its worse to have no backup than to have unnecessary change alerts.

> Unless this is a common occurrence, I don't think this warrants any extra code.

For me thats just standard error handling.

> If you insist on keeping it, the logging configuration will need to be documented/added to the project's default settings.

Yes, I think putting it into the default settings is a good idea.

> The config you suggested isn't terribly useful, since it would override the default config (which sends a mail each time an unhandled exception occurs).

With ```'disable_existing_loggers': False``` the default config from django should not get overwritten.

Reply:
>> But that has negative consequences, too - unnecessary commits will be made and the operations team gets spammed with change alerts.
>> 
>> I'd rather have intermittent failures than intermittent change alerts.
> 
> I think its worse to have no backup than to have unnecessary change alerts.

I'm with Alex on this one. If its possible we should always make a backup even if it generates lots of commits and spams ops. IMO a working Backup is the most important feature.

Reply:
The reason I'm cautious about that one is that so far, the main error handling approach was "let it fail and retry with a clean state"[1]. This is the first occasion in the project where a task catches, logs and ignores a runtime error. This introduces more complexity, and I wanted to make sure that it's warranted. Bypassing the retry logic also means that checksum comparisons will not be retried and result in a commit, even if they had otherwise succeeded on the second try. Given the CPU load issues on some devices, this is not an unlikely scenario.

I'm fine with keeping it. Please document the behavior for future users.

[1]: http://wiki.c2.com/?LetItCrash